src/Raven.Server/Documents/Indexes/IndexStorage.cs

```diff
@@ -536,12 +536,12 @@ public unsafe void WriteReferences(CurrentIndexingScope indexingScope, RavenTran
                         using (Slice.From(tx.InnerTransaction.Allocator, collectionName.Name, ByteStringType.Immutable, out Slice collectionKey))
                         {
                             var etag = collections.Value;
-
                             var result = collectionEtagTree.Read(collectionKey);
                             var oldEtag = result?.Reader.ReadLittleEndianInt64();
                             if (oldEtag >= etag)
                                 continue;
-
+                            if (oldEtag < etag)
+                                etag = oldEtag.Value;
```

Comment:
Not sure that I follow?

This is always 'true', I think that you will never increase the value, so you it will always process the same references, all over again.

Reply:
Its always oldEtag = etag on 64bit and on 32bit 
but if the index batch is splitted (because of the 8mb limitation) then oldEtag is the last etag of the document which will be indexed next, but if there no check for (oldEtag < etag) then etag just overwrite oldEtag and we miss indexing few documents.

Reply:
Let's add a comment here

Reply:
Fix here: https://github.com/ravendb/ravendb/pull/7728